### ISSUE TITLE
[Bug] Media Notifications aren't displayed 

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#5290]
## Steps to reproduce
1. Navigate to Spotify and play a song
2. Lock the device screen
3. Press the power button an check the lockscreen

### Expected behavior
Media notification should be displayed

### Actual behavior
No media notification is displayed
### Device information

* Android device: Samsung Galaxy Tab A (android 9)
* Fenix version: 2.0.0-rc.2

**Comment by ekager** [comment]
Is this the same as #5177 ?

**Comment by pocmo** [comment]
Possible. Let's see after #5177 is fixed.

**Comment by pocmo** [comment]
#5177 was fixed and the fix should be in Nightly. This can be retested now.

**Comment by sv-ohorvath** [comment]
Still no media notification from spotify.com on the latest Nightly 10/8 build.

**Comment by pocmo** [comment]
Thanks! I'll continue debugging this.

**Comment by pocmo** [comment]
Okay, how do I actually play something on spotify in the browser? On desktop I can go to open.spotify.com - but that doesn't seem to work when using a mobile browser?

**Comment by sv-ohorvath** [comment]
@pocmo I just logged in, picked a playlist from the homepage and tapped a song. You don't have the shuffle play option like on desktop though.

**Comment by sv-ohorvath** [comment]
@vesta0  this appears to be included in the 2.2 milestone but it's not done yet.

**Comment by softvision-miralobontiu** [comment]
Reproducible on RC 2.3.0-rc.2 build on Xiaomi Mi Pad 2 (Android 5.1) - x86-64 architecture.

**Comment by ekager** [comment]
Just retested this. I'm not seeing any notification displayed for Spotify. (Neither in the toolbar or on the homescreen)
STR:
1. Go to open.spotify.com. 
2. Ignore their "Open in Spotify" buttons and just click one of the albums to play a song

**Comment by softvision-miralobontiu** [comment]
I reproduced this issue playing videos from bbc.com, cnn.com, vimeo.com, on the latest Beta 80.0.0-beta.4 build, with Samsung Galaxy Tab S6 (Android 9), and OnePlus A3 (Android 6.0.1).
* Not reproducible on the latest Nightly build from 8/7.

**Comment by pocmo** [comment]
There will be a new MediaControl/MediaSession API in GeckoView soon. Android Components will switch to that and that should hopefully resolve a lot of issues. In the meantime we probably won't do any changes to the existing code anymore since that will probably all go away.
https://bugzilla.mozilla.org/show_bug.cgi?id=1623715

**Comment by dbolter** [comment]
FYI I happened to notice 1623715 landed so this should be unblocked. (Note: I'm not making any claims as to priority here)

**Comment by pocmo** [comment]
We are currently working on integrating the new GV API into AC.

**Comment by sv-ohorvath** [comment]
Still reproducing no notification of media playing shown in the status bar or in tabs tray, with sample videos from:
http://techslides.com/sample-webm-ogg-and-mp4-video-files-for-html5
https://file-examples.com/index.php/sample-video-files/

**Comment by pocmo** [comment]
Okay, this is now definitely the new API. :)

@rocketsroger: Can you investigate? Maybe we need to open a GV bug if we do not get anything for those sites.

**Comment by rocketsroger** [comment]
> Okay, this is now definitely the new API. :)
> 
> @rocketsroger: Can you investigate? Maybe we need to open a GV bug if we do not get anything for those sites.

Sure, I'll look into this.

**Comment by rocketsroger** [comment]
Yes, I can reproduce the issue with https://file-examples.com/index.php/sample-video-files/
But I can't reproduce for http://techslides.com/sample-webm-ogg-and-mp4-video-files-for-html5.  Media notification is shown. (short time since the video is short)

**Comment by rocketsroger** [comment]
https://file-examples.com/index.php/sample-video-files/ is an interesting case.  On both Chrome and Sample browser this results in a downloading of the file selected.  But on Fenix we play it directly.  Not sure if this is an expected behaviour.  I'll investigate more.

**Comment by sv-ohorvath** [comment]
> Yes, I can reproduce the issue with https://file-examples.com/index.php/sample-video-files/
> But I can't reproduce for http://techslides.com/sample-webm-ogg-and-mp4-video-files-for-html5. Media notification is shown. (short amount of time since the video is short)

You're right, I see the notification from techslides.com as well now. The same video formats (which play in the browser, not downloading) on https://www.learningcontainer.com/mp4-sample-video-files-download/ work. It might be something with the samples on https://file-examples.com/index.php/sample-video-files/

**Comment by rocketsroger** [comment]
Confirmed with GeckoView team that this is an expected behaviour.  In the https://file-examples.com/index.php/sample-video-files/ test case, except for AVI (which triggers a download) the other formats such as MOV and MP4 starts streaming.  GV does not notify A-C in this case as MediaSession hence we don't have a notification.

I'll get a Bugzilla issue to link here.

**Comment by rocketsroger** [comment]
Raised an issue for the file-example.com test case.  https://github.com/mozilla-mobile/android-components/issues/9748
Since the original issue here was Spotify and it now works correctly.  I'll close this issue and we can track the new issue to avoid confusion.
